python/python/lance/torch/kmeans.py

```diff
@@ -0,0 +1,140 @@
+#  Copyright (c) 2023. Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import Optional, Union
+
+import numpy as np
+import pyarrow as pa
+import torch
+from tqdm import tqdm
+
+from . import preferred_device
+from .distance import cosine_distance, dot_distance, l2_distance
+
+__all__ = ["KMeans"]
+
+
+@torch.jit.script
+def _random_init(
+    data: torch.Tensor, n: int, seed: Optional[int] = None
+) -> torch.Tensor:
+    if seed is not None:
+        torch.random.manual_seed(seed)
+    sample_idx = torch.randint(0, data.shape[0], (n,))
+    samples = data[sample_idx]
+    return samples
+
+
+class KMeans:
+    def __init__(
+        self,
+        k: int,
+        *,
+        metric: str = "l2",
+        redo: int = 2,
+        init_method: str = "random",
+        max_iters: int = 50,
+        tolerance: float = 1e-6,
+        seed: Optional[int] = None,
+        device: Optional[str] = None,
+    ):
```

Comment:
Could you add a docstring describing the parameters?

Reply:
Done